backend/src/main/java/com/lab/spring/backend/controllers/UserController.java

```diff
@@ -28,7 +29,7 @@ User getUserById(@Argument UUID id) {
     }
 
     @MutationMapping
-    User addUser(@Argument UserInput userInput) {
+    User addUser(@Argument @Valid UserInput userInput) {

```

Comment:
Sorry Trung lần trước a k để ý chỗ này. 
Mình đang sử dụng graphql tại sao lại dùng Controller và implement RestController 
Bình thường khi làm việc với graphql, để xử lý mutation và query chắc mình sẽ dùng Resolver và trong java spring boot nó có hỗ trợ thằng GraphQLQueryResolver. Resolver của mình sẽ implement thằng này chứ nhỉ. 

https://connect2grp.medium.com/using-graphql-with-spring-boot-java-application-c0b23cddbf4b

Reply:
@buiquangbk90 ở đây e dùng thằng [Spring for GraphQL](https://spring.io/guides/gs/graphql-server) do trang chủ Spring nó recommend ạ. Controller sẽ kết hợp với @MutationMapping và @QueryMapping đóng vai trò như Resolver thôi ạ, e implement @Controller là đủ ạ nhưng e chưa xóa code cũ dùng Rest nên hơi thừa ạ.
Link a gửi nó dùng thư viện [GraphQL Java](https://www.graphql-java.com/) nguyên thủy ạ, Spring đã build lại thư viện dựa trên GraphQL Java để đỡ phải customize nhiều ạ.